Sources/MapboxCoreNavigation/TileStoreLocation.swift

```diff
@@ -0,0 +1,64 @@
+
+import Foundation
+import MapboxCommon
+
+/**
+ To add docs for cases
+ */
+public enum TileStoreLocation {
```

Comment:
I have an idea for this class:
```swift
public struct TileStoreConfiguration {
    enum Location: RawRepresentable {
        case `default`
        case custom(URL)

        init?(rawValue: URL?) {
            self = rawValue.map { .custom($0) } ?? .default
        }

        var rawValue: URL? {
            switch self {
            case .default:
                return nil
            case .custom(let url):
                return url
            }
        }

        var tileStore: TileStore {
            switch self {
            case .default:
                return TileStore.getInstance()
            case .custom(let url):
                return TileStore.getInstanceForPath(url.path)
            }
        }
    }

    var navigatorLocation: Location
    var mapLocation: Location?

    static var `default`: Self {
        .init(navigatorLocation: .default, mapLocation: .default)
    }
    static func custom(_ url: URL) -> Self {
        .init(navigatorLocation: .custom(url), mapLocation: .custom(url))
    }
    static func isolated(navigationLocation: Location, mapLocation: Location?) -> Self {
        .init(navigatorLocation: navigationLocation, mapLocation: mapLocation)
    }
}
```

Reply:
I like this! I think I'll try this in next iteration

Reply:
I'm not sure if `Location` enum needs `RawRepresentable`, but it looks cooler :) But also this makes some ambiguity for `let x = Location(rawValue: nil)` vs `let x: Location? = nil`.